code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/InstancePurgePostListener.java

```diff
@@ -0,0 +1,63 @@
+package io.cattle.platform.servicediscovery.process;
+
+import static io.cattle.platform.core.model.tables.InstanceHostMapTable.INSTANCE_HOST_MAP;
+
+import io.cattle.platform.allocator.service.CacheManager;
+import io.cattle.platform.allocator.service.HostInfo;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.core.model.InstanceHostMap;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPostListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.servicediscovery.api.dao.ServiceDao;
+import io.cattle.platform.util.type.Priority;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InstancePurgePostListener extends AbstractObjectProcessLogic implements ProcessPostListener, Priority {
+
+    private static final Logger log = LoggerFactory.getLogger(InstancePurgePostListener.class);
+
+    @Inject
+    ServiceDao serviceDao;
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { InstanceConstants.PROCESS_PURGE };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        Instance instance = (Instance) state.getResource();
+
+        List<InstanceHostMap> instanceHostMappings = objectManager.find(InstanceHostMap.class,
+                INSTANCE_HOST_MAP.INSTANCE_ID,
+                instance.getId());
+
+        if (instanceHostMappings.size() == 0) {
+            return null;
+        }
+        InstanceHostMap mapping = instanceHostMappings.get(0);
```

Comment:
Why just the first one? I realize that logically there will only be one, but I think you should just throw this in a loop and do it for each InstanceHostMap that is returned.